larpandoracontent/LArHelpers/LArHierarchyHelper.cc

```diff
@@ -0,0 +1,1171 @@
+        {
+            return lhs->GetCaloHits().size() > rhs->GetCaloHits().size();
+        });
+    std::sort(recoNodes.begin(), recoNodes.end(), [](const RecoHierarchy::Node *lhs, const RecoHierarchy::Node *rhs)
+        {
+            return lhs->GetCaloHits().size() > rhs->GetCaloHits().size();
+        });
+
+    std::map<const MCHierarchy::Node *, MCMatches> mcToMatchMap;
+    for (const RecoHierarchy::Node *pRecoNode : recoNodes)
+    {
+        const CaloHitList &recoHits{pRecoNode->GetCaloHits()};
+        const MCHierarchy::Node *pBestNode{nullptr};
+        size_t bestSharedHits{0};
+        for (const MCHierarchy::Node *pMCNode : mcNodes)
+        {
+            if (!pMCNode->IsReconstructable())
+                continue;
+            const CaloHitList &mcHits{pMCNode->GetCaloHits()};
+            CaloHitVector intersection;
+            std::set_intersection(mcHits.begin(), mcHits.end(), recoHits.begin(), recoHits.end(), std::back_inserter(intersection));
+
+            if (!intersection.empty())
+            {
+                const size_t sharedHits{intersection.size()};
+                if (sharedHits > bestSharedHits)
+                {
+                    bestSharedHits = sharedHits;
+                    pBestNode = pMCNode;
+                }
+            }
+        }
+        if (pBestNode)
+        {
+            auto iter{mcToMatchMap.find(pBestNode)};
+            if (iter != mcToMatchMap.end())
+            {
+                MCMatches &match(iter->second);
+                match.AddRecoMatch(pRecoNode, static_cast<int>(bestSharedHits));
+            }
+            else
+            {
+                MCMatches match(pBestNode);
+                match.AddRecoMatch(pRecoNode, static_cast<int>(bestSharedHits));
+                mcToMatchMap.insert(std::make_pair(pBestNode, match));
+            }
+        }
+        else
+        {
+            m_unmatchedReco.emplace_back(pRecoNode);
+        }
+    }
+
+    for (auto [ pMCNode, matches ] : mcToMatchMap)
+    {   (void)pMCNode;  // gcc 7 requirement
+        const RecoHierarchy::NodeVector &nodeVector{matches.GetRecoMatches()};
+        if (nodeVector.size() == 1)
+        {
+            const RecoHierarchy::Node *pRecoNode{nodeVector.front()};
+            const float purity{matches.GetPurity(pRecoNode)};
+            const float completeness{matches.GetCompleteness(pRecoNode)};
+            if (purity >= m_qualityCuts.m_minPurity && completeness > m_qualityCuts.m_minCompleteness)
+                m_goodMatches.emplace_back(matches);
+            else
+                m_subThresholdMatches.emplace_back(matches);
+        }
+        else
+        {
+            m_subThresholdMatches.emplace_back(matches);
+        }
+    }
+
+    auto predicate = [](const MCMatches &lhs, const MCMatches &rhs)
+        {
+            return lhs.GetMC()->GetCaloHits().size() > rhs.GetMC()->GetCaloHits().size();
+        };
```

Comment:
Over-indented?

Reply:
We've started using a .clang-format file to standardise formatting for all future PRs. So this is a clang choice. Hopefully once a newer version of clang is available on CVMFS we'll have more flexibility to fix some of the things we don't like.

Reply:
Bit of a clanger? 
Sorry.